internal/core/domain/vault_model.go

```diff
@@ -37,30 +38,98 @@ type AddressInfo struct {
 	DerivationPath string
 }
 
+// IMnemonicStore defines the required methods to override the default
+// storage of the plaintext mnemonic once Unlocking a Vault.
+// At the moment this is achieved by storing the mnemonic in the in-memory
+// pkg/config store. Soon this will be changed since this package shouldn't
+// depend on config.
+type IMnemonicStore interface {
+	Set(mnemonic string)
+	Unset()
+	IsSet() bool
+	Get() []string
+}
+
+// TOOD: move this outside the domain to decouple from config and use an
+// equivalent alternative.
+type configStore struct{}
+
+func (c configStore) Set(mnemonic string) {
+	config.Set(config.MnemonicKey, mnemonic)
+}
+
+func (c configStore) Unset() {
+	config.Set(config.MnemonicKey, nil)
+}
+
+func (c configStore) IsSet() bool {
+	return len(config.GetMnemonic()) > 0
+}
+
+func (c configStore) Get() []string {
+	return config.GetMnemonic()
+}
+
+// IEncrypter defines the required methods to override the default encryption
+// performed through pkg/wallet package.
+type IEncrypter interface {
+	Encrypt(mnemonic, passphrase string) (string, error)
+	Decrypt(encryptedMnemonic, passphrase string) (string, error)
+}
+
+type walletEncrypter struct{}
+
+func (w walletEncrypter) Encrypt(mnemonic, passphrase string) (string, error) {
```

Comment:
here we are just wrapping wallet Encrypt and Decrypt methods and creating interfaces, i guess goal was to mock Encrypt and Decrypt methods in tests, i think exposing of an interface was supposed to be done in wallet package itself

Reply:
Not saying one way is better that other, but style changes should be done in a dedicated PR, not mixed with functional/testing stuff

Reply:
Encrypt and Decrypt are static functions of the wallet package. I think the encrypter interafce is more related to the domain rather than the wallet package itself. After all, it's required by the Vault domain and it's there that we need to define how the encrypter should look like from a domain-level point of view.